libs/feature/record/src/lib/service/data.service.ts

```diff
@@ -0,0 +1,150 @@
+import { Injectable } from '@angular/core'
+import { from, Observable, throwError } from 'rxjs'
+import { WfsEndpoint } from '@camptocamp/ogc-client'
+import { MetadataLinkValid, ProxyService } from '@geonetwork-ui/util/shared'
+import { catchError, map, pluck, tap } from 'rxjs/operators'
+import type { FeatureCollection } from 'geojson'
+import { readDataset, SupportedType } from '@geonetwork-ui/data-fetcher'
+import { marker } from '@biesbjerg/ngx-translate-extract-marker'
+
+marker('wfs.unreachable.cors')
+marker('wfs.unreachable.http')
+marker('wfs.unreachable.unknown')
+marker('wfs.featuretype.notfound')
+marker('wfs.geojson.notsupported')
+marker('dataset.error.network')
+marker('dataset.error.http')
+marker('dataset.error.parse')
+marker('dataset.error.unknown')
+
+interface WfsDownloadUrls {
+  all: { [format: string]: string }
+  geojson: string
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DataService {
+  constructor(private proxy: ProxyService) {}
+
+  private getDownloadUrlsFromWfs(
+    wfsUrl: string,
+    featureTypeName: string
+  ): Observable<WfsDownloadUrls> {
+    return from(
+      new WfsEndpoint(this.proxy.getProxiedUrl(wfsUrl)).isReady()
+    ).pipe(
+      catchError((error) => {
+        if (error.isCrossOriginRelated) {
+          throw new Error(`wfs.unreachable.cors`)
+        } else if (error.httpStatus) {
+          throw new Error(`wfs.unreachable.http`)
+        } else {
+          throw new Error(`wfs.unreachable.unknown`)
+        }
+      }),
+      // eslint-disable-next-line -- ogc-client lacks typing
+      map((endpoint: any) => {
+        const featureType = endpoint.getFeatureTypeSummary(featureTypeName)
+        if (!featureType) {
+          throw new Error('wfs.featuretype.notfound')
+        }
+        return {
+          all: featureType.outputFormats.reduce(
+            (prev, curr) => ({
+              ...prev,
+              [curr]: endpoint.getFeatureUrl(featureType.name, {
+                outputFormat: curr,
+              }),
+            }),
+            {}
+          ),
+          geojson: endpoint.supportsJson(featureType.name)
+            ? endpoint.getFeatureUrl(featureType.name, {
+                asJson: true,
+                outputCrs: 'EPSG:4326',
+              })
+            : null,
+        }
+      })
+    )
+  }
+
+  getGeoJsonDownloadUrlFromWfs(
+    wfsUrl: string,
+    featureType: string
+  ): Observable<string> {
+    return this.getDownloadUrlsFromWfs(wfsUrl, featureType).pipe(
+      pluck('geojson'),
```

Comment:
Didn't know this one, but looking in the docs, it looks deprecated. Maybe use `map` here?

Reply:
good catch, thanks!